src/test/java/com/icthh/xm/ms/balance/service/BalanceServiceUnitTest.java

```diff
@@ -1825,4 +1832,395 @@ public void setIsLastBalanceChangeEvent() {
+            .thenReturn(List.of(existBalanceChangeEvent));
+        when(balanceChangeEventRepository.existsByRevertOperationId(uuidToRevert))
+            .thenReturn(false);
+        when(pocketRepository.findOneByIdForUpdate(123L)).thenReturn(of(pocket));
+
+        balanceService.revertBalanceOperation(revertBalanceOperationRequest);
+    }
+
+    @Test
+    public void shouldSuccessWhenRevertReloadAsManyAsPossibleFromPocketsAndChargeToNegativeRemainder() {
+        long balanceId = 123456789;
+        String currentUuid = UUID.randomUUID().toString();
+        String uuidToRevert = UUID.randomUUID().toString();
+        long pocketId = 12345L;
+
+        RevertBalanceOperationRequest revertBalanceOperationRequest = new RevertBalanceOperationRequest();
+        revertBalanceOperationRequest.setBalanceId(balanceId);
+        revertBalanceOperationRequest.setUuid(currentUuid);
+        revertBalanceOperationRequest.setUuidToRevert(uuidToRevert);
+        revertBalanceOperationRequest.setRevertReloadMode(RevertReloadMode.AS_MANY_AS_POSSIBLE_FROM_POCKETS);
+        when(balanceChangeEventRepository.findBalanceChangeEventsByOperationId(revertBalanceOperationRequest.getUuid()))
+            .thenReturn(emptyList());
+
+        BalanceChangeEvent existBalanceChangeEvent = createBalanceEvent("90", balanceId, RELOAD, "100", "10");
+        existBalanceChangeEvent.setOperationId(uuidToRevert);
+        existBalanceChangeEvent.setPocketChangeEvents(List.of(
+            createPocketEvent("90", pocketId, "key", "label", "90", "0")
+        ));
+        when(balanceChangeEventRepository.findBalanceChangeEventsByOperationId(uuidToRevert))
+            .thenReturn(List.of(existBalanceChangeEvent));
+        when(balanceChangeEventRepository.existsByRevertOperationId(uuidToRevert))
+            .thenReturn(false);
+
+        createBalance(balanceId);
+        expectedAuth();
+        when(balanceRepository.findBalanceAmount(any(), any())).thenReturn(of(new BigDecimal(30)));
+
+
+        Pocket pocket = new Pocket().key("ASSERTION_KEY").label("label").amount(new BigDecimal("30"));
+        pocket.setId(pocketId);
+        when(pocketRepository.findOneByIdForUpdate(pocketId)).thenReturn(of(pocket));
+        when(pocketRepository.save(refEq(pocket))).thenReturn(pocket);
+
+        balanceService.revertBalanceOperation(revertBalanceOperationRequest);
+
+        BalanceChangeEvent expectedBalanceChangeEvent = BalanceChangeEvent.builder()
+            .amountDelta(new BigDecimal(30))
+            .operationType(REVERT_RELOAD)
+            .operationId(currentUuid)
+            .amountBefore(new BigDecimal(30))
+            .amountAfter(new BigDecimal(0))
+            .balanceId(balanceId)
+            .amountTotal(new BigDecimal(90))
+            .revertOperationId(uuidToRevert)
+            .build();
+
+        verify(balanceChangeEventRepository).save(
+            refEq(expectedBalanceChangeEvent,
```

Comment:
try to use
ArgumentMathers.argThat
it will be better in this case

Reply:
example
` argThat(reloadBalanceDTO -> 
            (reloadBalanceDTO.balanceId == balanceId
            && reloadBalanceDTO.amount == amount
            && reloadBalanceDTO.label == entityId.toString()
            && reloadBalanceDTO.uuid == operationKey)
        )`

Reply:
Fixed